Providers/PocketFeed.swift

```diff
@@ -48,11 +48,16 @@ private class PocketError: MaybeErrorType {
 }
 
 class Pocket {
-
+    private let consumerKey: String
     private let pocketGlobalFeed: String
+    
     // Allow endPoint to be overriden for testing
-    init(endPoint: String? = nil) {
-        pocketGlobalFeed = endPoint ?? PocketGlobalFeed
+    init?(endPoint: String? = nil) {
+        guard let consumerKey = Bundle.main.object(forInfoDictionaryKey: PocketEnvAPIKey) as? String else {
+            return nil
```

Comment:
I think this breaks the tests. For tests, I run a fake api endpoint that returns pocket stories without an API key. This is also why I let you provide your `endPoint`

Reply:
I think the difference between my pattern in yours is that I just return an empty array slightly later. I dont think there is any significant processing that happens before that. So if the API decides to return nil at the last step after its created a url, calculated the locale, checked a pref I think its okay. 

The thing I was trying to go for from the ASPanel frontend perspective was that the error handling would only happen in _one_ place. Instead of checking if the pocketAPI is nil or that localIdentifier is okay I create the Deferred and if any of those cases occur I just handle the empty array there (granted, I should handle the errors better).